aiida/orm/utils/calcjob.py

```diff
@@ -97,17 +97,17 @@ def __getattr__(self, name):
         """
         try:
             return self.get_results()[name]
-        except AttributeError:
+        except KeyError:
```

Comment:
I think we need to guard also for any exception that can happen in `get_results`, both here and below.

I found at least two possibilities, both a `ValueError`:
```
ValueError: cannot load results because process class cannot be loaded: no process type for CalcJobNode<None>: cannot recreate process class

ValueError: cannot load results as the default node could not be retrieved: no neighbor with the label output_parameters found
```
I am not sure if there is still some case if an AttributeError can still be raised.

Reply:
For tests, in the first case I just created an unstored `CalcJobNode()`; in the second one, a calculation that excepted and hand no output node.